src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/AdjournmentCalculateDateHelper.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.sscs.helper.mapping;
+
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.beanutils.BeanUtils;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.Optional;
+import javax.validation.Valid;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+@Slf4j
+public final class AdjournmentCalculateDateHelper {
+
+    static final int DAYS_TO_ADD_HEARING_WINDOW_TODAY_ADJOURNMENT = 14;
+
+    private AdjournmentCalculateDateHelper() {
+    }
+
+    static LocalDate getHearingWindowStart(@Valid SscsCaseData caseData) {
+        checkForMissingData("adjournCaseNextHearingDateType", caseData);
+        // TODO add to "Other" field time details from adjournCaseTime
+        String adjournCaseNextHearingDateType = caseData.getAdjournCaseNextHearingDateType();
+        Optional<AdjournmentDateType> adjournmentDateType = findDateTypeByValue(adjournCaseNextHearingDateType);
+        if (adjournmentDateType.isPresent()) {
```

Comment:
You should be able to flip these around to make them a bit neater. Using Optional's orElseThrow you can do something like:

```suggestion
        AdjournmentDateType adjournmentDateType =
            AdjournmentDateType.findDateTypeByValue(adjournCaseNextHearingDateType)
            .orElseThrow(() -> new IllegalArgumentException(String.format(
                "Unexpected adjournCaseNextHearingDateType: %s for case id %s",
                adjournCaseNextHearingDateType,
                caseData.getCcdCaseId())));

        return adjournmentDateType.resolveDate(caseData);
```

Reply:
Ah nice, I was trying to roll it all into a single return statement but couldn't figure out how to squeeze the error in. Saving it to a temp variable is the way, thanks.
Done